sat/report.py

```diff
@@ -159,19 +158,23 @@ def __init__(self, headings, title=None,
 
         # find the heading to sort on
         if sort_by is not None:
+            if not isinstance(self.sort_by, list):
+                self.sort_by = [self.sort_by]
             warn_str = "Element '%s' is not in %s. Output will be unsorted."
-            try:
-                self.sort_by = int(self.sort_by)
-                self.sort_by = self.headings[self.sort_by]
-            except IndexError:
-                # sort_by is out of range.
-                LOGGER.warning(warn_str, sort_by, self.headings)
-                self.sort_by = None
-            except ValueError:
-                # sort_by is not an int.
-                self.sort_by = match_query_key(self.sort_by, headings)
-                if not self.sort_by:
+            for i in range(len(self.sort_by)):
+                try:
+                    self.sort_by[i] = int(self.sort_by[i])
+                    self.sort_by[i] = self.headings[self.sort_by[i]]
+                except IndexError:
+                    # sort_by is out of range.
                     LOGGER.warning(warn_str, sort_by, self.headings)
+                    self.sort_by = None
```

Comment:
We might want to consider ignoring only the `sort_by` elements that are not valid, rather than ignoring all of them. In that case, you might need to generate a new, converted list in this for loop. For invalid elements of `sort_by`, you would just skip adding them to that new list with a logged warning. Then at the end of the loop, you'd set `self.sort_by` to this new list. If the new list is empty, then it means none of the given `sort_by` values were valid, so then you'd set `self.sort_by = None` and log the warning that the output will be unsorted.

This is a pretty minor edge case though. I could see it maybe being useful for the commands we have that actually print multiple reports (e.g. `sat showrev`) where those reports have different headings. E.g., the `--products` output vs. the `--local` output have different headings, and a comma-separated `--sort-by` value could be used to sort the two different reports by different headings.